modules/apps/29-fee/ibc_middleware.go

```diff
@@ -218,12 +218,13 @@ func (im IBCMiddleware) OnRecvPacket(
 	ctx sdk.Context,
 	packet channeltypes.Packet,
 	relayer sdk.AccAddress,
+	channelVersion string,
 ) exported.Acknowledgement {
 	if !im.keeper.IsFeeEnabled(ctx, packet.DestinationPort, packet.DestinationChannel) {
-		return im.app.OnRecvPacket(ctx, packet, relayer)
+		return im.app.OnRecvPacket(ctx, packet, relayer, channelVersion)
 	}
 
-	ack := im.app.OnRecvPacket(ctx, packet, relayer)
+	ack := im.app.OnRecvPacket(ctx, packet, relayer, channelVersion)
```

Comment:
we need to be unwrapping the version. This should break if you have a transfer <> ics29 <> coreIBC stack with fee enabled as the fee version would be given to transfer

The logic to unwrap is in `GetAppVersion`

Reply:
I have been trying to figure out how to reproduce what you describe to make sure I get this implemented properly, but I don't think I understand exactly how to set up for this. Why doesn't the e2e tests fail, are there any other ways of setting this up than: 
```go
var transferStack porttypes.IBCModule
transferStack = transfer.NewIBCModule(app.TransferKeeper)
transferStack = ibcfee.NewIBCMiddleware(transferStack, app.IBCFeeKeeper)
```